serial.ts

```diff
@@ -244,7 +244,7 @@ export class SerialPort {
                 this.readable_ = null;
               }),
           new ByteLengthQueuingStrategy({
-            highWaterMark: this.serialOptions_.bufferSize ?? kDefaultBufferSize,
+              highWaterMark: this.serialOptions_.bufferSize ?? this.inEndpoint_.packetSize > kDefaultBufferSize ? this.inEndpoint_.packetSize : kDefaultBufferSize,
```

Comment:
This isn't the most reliable way of fixing the issue because if there is room in the queue then `controller.desiredSize` may still return less than the maximum packet size inside `pull()`. Instead can you modify `pull()` to  take `controller.desiredSize` and round it up to the nearest multiple of the maximum packet size?

Reply:
pull(controller: ReadableStreamDefaultController<Uint8Array>): void {
        (async (): Promise<void> => {
            let chunkSize;
            if (controller.desiredSize) {
                if (controller.desiredSize < this.endpoint_.packetSize) {
                    chunkSize = this.endpoint_.packetSize;
                } else {
                    chunkSize = Math.ceil(this.endpoint_.packetSize / controller.desiredSize) * this.endpoint_.packetSize;
                }
            } else {
                chunkSize = this.endpoint_.packetSize;
            } 

you think something like this would do?

Reply:
I like the version using if/else better than the version where it is all combined into a single expression. The only issue is that I think I needs to be `controller.desiredSize / this.endpoint_.packetSize` so that, for example, if the desired size is 100 bytes but the maximum packet size is 64 bytes the division is rounded up to 2 and we get a chunk size of 128 bytes.

Reply:
my mistake, "controller.desiredSize / this.endpoint_.packetSize" obviously

changed it.

Asuming that if desiredSize and packetSize exist and they are both > 0  I think this could also be written like this:

    if (controller.desiredSize) {
      const d = controller.desiredSize / this.endpoint_.packetSize;
      chunkSize = Math.ceil(d) * this.endpoint_.packetSize;
    } else {
      chunkSize = this.endpoint_.packetSize;
    }

in the conditional
let a be controller.desiredSize
and b be this.endpoint_.packetSize
let a>0 and b>0
if a<b then the division a/b would be in (0,.,1) and ceil would be 1
if a>=b then a/b would be in [1,..r) and ceil would be in [1,..,n]

should we use that one?

Reply:
I agree, the extra check comparing desiredSize and packetSize is unnecessary as ceil will round up to 1.